.circleci/config.yml

```diff
@@ -94,12 +94,11 @@ jobs:
     machine: true
     steps:
       - checkout
-      - run: docker build --network host --build-arg=EMSCRIPTEN_VERSION=${CIRCLE_TAG}-upstream --tag emscripten/emsdk:${CIRCLE_TAG} ./docker
-      - run: docker tag emscripten/emsdk:${CIRCLE_TAG} emscripten/emsdk:latest
+      - run: make -C ./docker version=${CIRCLE_TAG} build
+      - run: make -C ./docker version=${CIRCLE_TAG} test
       - run: |
           docker login -u "$DOCKER_USER" -p "$DOCKER_PASS"
-          docker push emscripten/emsdk:${CIRCLE_TAG}
-          docker push emscripten/emsdk:latest
+          make -C ./docker version=${CIRCLE_TAG} alias=latest push
```

Comment:
Could we split this into two jobs `build-docker-image` and `publish-docker-image` and have `build-docker-image` run always with no filter.. so we can build and test the image on all PRs?

Reply:
(No need to make this part of this PR .

Reply:
Sounds good, that will require some experiments to make sure it works like intended, let's make another PR for that.

Reply:
I can take care of that if you like.. since it would be nice to verify that docker build and test as part of my PR and this one.